api/permissions/enforcer.go

```diff
@@ -56,3 +62,40 @@ func Init(config viper.Viper) (*casbin.Enforcer, error) {
 
 	return e, nil
 }
+
+// IPMultipleMatch determines whether any of IP address in ip1 matches the pattern of any IP address in ip2, ip2 can be an IP address or a CIDR pattern.
+func IPMultipleMatch(ips1 string, ips2 string) bool {
+	anyMatch := false
+	for _, ip2 := range strings.Split(ips2, ";") {
+		for _, ip1 := range strings.Split(ips1, ";") {
+			objIP1 := net.ParseIP(ip1)
+			if objIP1 == nil {
+				panic("invalid argument: ip1 in IPMultipleMatch() function is not an IP address.")
+			}
+			_, cidr, err := net.ParseCIDR(ip2)
+			if err != nil {
+				objIP2 := net.ParseIP(ip2)
+				if objIP2 == nil {
+					panic("invalid argument: ip2 in IPMultipleMatch() function is neither an IP address nor a CIDR.")
+				}
+
+				if objIP1.Equal(objIP2) == true {
+					anyMatch = true
+				}
+			}
+
+			if cidr.Contains(objIP1) == true {
+				anyMatch = true
+			}
+		}
+	}
+	return anyMatch
+}
+
+// IPMultipleMatchFunc is the wrapper for IPMultipleMatch.
+func IPMultipleMatchFunc(args ...interface{}) (interface{}, error) {
+	ip1 := args[0].(string)
+	ip2 := args[1].(string)
+
```

Comment:
Do you think it would be a good idea checking if both these arguments are valid? 🙃

Is the following case possible to happen?
`args[0] = 10.10.10.2` and `args[1] = ""`

Reply:
This function call `IPMultipleMatch`. `IPMultipleMatch` checks for validity of `ip1` and `ip2` using `net.ParseIP()` function.

If `args[0] = 10.10.10.2 and args[1] = ""` happens, the `IPMultipleMatch` function handles this returning an error (after panic - this is a cabin limitation).